lib/smart_answer/erb_renderer.rb

```diff
@@ -14,22 +14,22 @@ def initialize(template_directory:, template_name:, locals: {}, helpers: [])
       @template_directory = template_directory
       @template_name = template_name
       @locals = locals
-      lookup_context = ActionView::LookupContext.new([@template_directory, FlowRegistry.instance.load_path])
+      default_view_paths = ActionController::Base.view_paths.paths.map(&:to_s)
```

Comment:
Can you help me understand this line?

Reply:
Yup - my understanding when rails initialises it creates a list of view paths (places to search for templates  e.g. "app/views") when calling methods like `render`. GovukPublishingComponents gem adds its own views paths to this list. 

So here we are grabbing the view paths ("app/view", "./path/to/gem/views") that would normally be available to ActionController (i.e. normal Rails controller context). ErbRenderer is making its own ActionView::Base object (the thing responsible for rendering templates etc.), separate to the one that ActionController uses. 

We need to give ActionView::Base a list of view paths (within LookupContext) when we create a new instance. We want to copy the view paths from what we normally have in the controller to our new ActionView::Base object. Ultimately we are trying get the GovukPublishingComponents gem's views paths for our new ActionView::Base object. Admittedly we are assuming that the gems paths have been added and blindly copying everything.

Context for that line specifically:

View paths are logically just paths (strings)  - but here we are just unwrapping them from Rails objects:
```ruby
path_set = ActionController::Base.view_paths # Gets a ActionView::PathSet object (the one to be used by the ActionView::Base associated with ActionController::Base i.e. normal Rails context)
paths = path_set.paths # Get an Array of ActionView::OptimizedFileSystemResolver objects
path_strings = paths.map(&:to_s) # Calling to_string method on each OptimizedFileSystemResolver objects (return the @path attribute)
```

Reply:
Ah, I see. Thanks for the explanation of the mechanism. I was just missing the part at the beginning: we want to support rendering GOV.UK Publishing Components. Can add this to [the commit message](https://github.com/alphagov/smart-answers/pull/4502/commits/0983a2353b4d5456fda2c38e6f478bb664624c7a)? It's somewhat tangential to enabling raw HTML.

Reply:
Yup - added to the[ commit message.](https://github.com/alphagov/smart-answers/pull/4502/commits/96bdfb4dd63cd1e6e813d48292b727fdc33f0725)